Sources/Player/Types/TrackingBehavior.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  Copyright (c) SRG SSR. All rights reserved.
+//
+//  License information is available from the LICENSE file.
+//
+
+/// A tracking behavior.
+///
+/// Determines how a ``TrackerAdapter`` is affected by the ``Player/isTrackingEnabled`` player setting.
+public enum TrackingBehavior {
+    /// Optional tracking.
+    ///
+    /// The ``TrackerAdapter`` takes into account the ``Player/isTrackingEnabled`` player setting.
+    case optional
+
+    /// Mandatory tracking.
+    ///
+    /// The ``TrackerAdapter`` ignores the ``Player/isTrackingEnabled`` player setting.
+    case mandatory
+}
```

Comment:
Isn't a boolean sufficient? (e.g. `isTrackingMandatory` or `isTrackingOptional`)

Reply:
This was my first iteration but it felt clunky IMHO. 

Moreover, and though I did not implement it, we could have additional behaviors like "tracking if the app is in the foreground only". This is something we discussed in the context of our analytics requirements and an enum can be easily extended with additional behaviors, unlike a Bool.

TL;DR I found it both more expressive and extensible, thus this proposal.

Reply:
Let me elaborate a bit more about the _clunkiness_ I was mentioning.

Booleans are great at describing simple self-describing binary states like `enabled` or `visible`, for example. They are less so at describing richer behaviors that cannot be described in simple terms, like `immediate` for our seeking behavior or the tracking behavior we are discussing. The tracking behavior is here not a simple state that can be understood in itself, it is merely a characterization of the interaction between a tracker behavior and how the parent player has been configured. It is only in response to this configuration that a tracker exhibits a certain behavior, it cannot be understood in isolation.

If you look at my first commit you can maybe see why this is clunky when a simple `mandatory` Boolean is used / read in code afterwards. If the Boolean is used as is (the `isTrackingMandatory` is probably too verbose, `mandatory` is the concept here since a tracker's purpose is to track), then you would have to add it everywhere this context is missing, because just seeing a _mandatory_ Boolean does not tell you what is mandatory (is it the thing you are setting? Is it another mandatory behavior like a user consent for example?). Thus each call site would likely need to be adorned with `isTrackingMandatory` or `isUserConsentMandatory` to provide for the missing context.

It is similar with simple words like `enabled`. While they can be great at conveying _light bulb_ states, they do not tell you which of these states is actually controlled. Once several states can have these on / off states, you need to provide context when they are mixed, otherwise you don't know which one is which.

Reply:
I see your point, thanks for the explanation.